scaleway/resource_rdb_user_beta.go

```diff
@@ -0,0 +1,176 @@
+package scaleway
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/rdb/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayRdbUserBeta() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayRdbUserBetaCreate,
+		Read:   resourceScalewayRdbUserBetaRead,
+		Update: resourceScalewayRdbUserBetaUpdate,
+		Delete: resourceScalewayRdbUserBetaDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"instance_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validationUUIDorUUIDWithLocality(),
+				Description:  "Instance on which the user is created",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Database user name",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+				Description: "Database user password",
+			},
+			"is_admin": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Grant admin permissions to database user",
+			},
+			// Common
+			"region": regionSchema(),
+		},
+	}
+}
+
+func resourceScalewayRdbUserBetaCreate(d *schema.ResourceData, m interface{}) error {
+	rdbAPI, region, err := rdbAPIWithRegion(d, m)
+	if err != nil {
+		return err
+	}
+	instanceId := d.Get("instance_id").(string)
+	createReq := &rdb.CreateUserRequest{
+		Region:     region,
+		InstanceID: expandID(instanceId),
+		Name:       d.Get("name").(string),
+		Password:   d.Get("password").(string),
+		IsAdmin:    d.Get("is_admin").(bool),
+	}
+
+	res, err := rdbAPI.CreateUser(createReq)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s", region, expandID(instanceId), res.Name))
+
+	return resourceScalewayRdbUserBetaRead(d, m)
+}
+
+func resourceScalewayRdbUserBetaRead(d *schema.ResourceData, m interface{}) error {
+	rdbAPI, region, err := rdbAPIWithRegion(d, m)
+	if err != nil {
+		return err
+	}
+
+	regionName, instanceId, userName, err := resourceScalewayRdbUserBetaParseId(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	res, err := rdbAPI.ListUsers(&rdb.ListUsersRequest{
+		Region:     region,
+		InstanceID: instanceId,
+		Name:       &userName,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	var user = res.Users[0]
+	_ = d.Set("instance_id", fmt.Sprintf("%s/%s", region, instanceId))
+	_ = d.Set("name", user.Name)
+	_ = d.Set("password", d.Get("password").(string)) // password are immutable
```

Comment:
Does that mean that password are stored in plain test ?

Reply:
Would it be better to hash this to avoid storing sensitive information in plain text? https://registry.terraform.io/providers/hashicorp/random/latest/docs/resources/password

Reply:
I am following recommendations from https://www.terraform.io/docs/state/sensitive-data.html
The password is stored as is in local state. However this is the standard behaviour for providers, user are encouraged to use an encrypted state store when using secrets.